File: Olimpiadi di informatica/grades/ois_grades/grades.pas
{ NOTE: it is recommended to use this even if you don't understand the following code. }

{ constraints }
const
    MAXN = 1000000;

{ input data }
var
    N, i  : longint;
    G     : array[0..MAXN-1] of longint;

begin
{
    uncomment the following lines if you want to read/write from files
    assign(input,  'input.txt');  reset(input);
    assign(output, 'output.txt'); rewrite(output);
}

    readln(N);
    for i:=0 to N-1 do
        read(G[i]);
    readln();

    { insert your code here }

    writeln(42); { print result }
end.

File: Olimpiadi di informatica/grades/grades.c
#include <stdio.h>
#include <assert.h>

// constraints
#define MAXN 1000000

// input data
int N, i;
int G[MAXN];
int risultato;

int main()
{
    //  uncomment the following lines if you want to read/write from files
    //  freopen("ois_grades/input0.txt", "r", stdin);
    //  freopen("output.txt", "w", stdout);

    assert(1 == scanf("%d", &N));
    for (i = 0; i < N; i++)
        assert(1 == scanf("%d", &G[i]));

    // Trovo la posizione del primo voto negativo
    int indexPrimoVotoNegativo = -1;

    for (i = 0; i < N && indexPrimoVotoNegativo == -1; i++)
        if (G[i] < 6)
            indexPrimoVotoNegativo = i;

    // Controllo se ho trovato almeno un numero negativo
    if (indexPrimoVotoNegativo >= 0)
    {
        // Cerco la posizione dell'ultimo voto positivo in lista
        int indexUltimoVotoPositivo = -1;
        for (i = N - 1; i > 0 && indexUltimoVotoPositivo == -1; i--)
            if (G[i] >= 6)
                indexUltimoVotoPositivo = i;

        // Controllo se è stato trovato un voto positivo
        if (indexUltimoVotoPositivo >= 0)
        {
            risultato = indexUltimoVotoPositivo - indexPrimoVotoNegativo + 1;

            if(risultato <= 0) risultato = -1;
        }
        else
        {
            risultato = -1;
        }
    }
    else
    {
        risultato = -1;
    }

    printf("%d\n", risultato); // print the result
    return 0;
}
